app/client/sagas/filterSaga.ts

```diff
@@ -51,9 +53,13 @@ function* watchLoadFilters() {
   yield takeEvery(LoadFilterActionTypes.LOAD_FILTERS_REQUEST, loadFilters);
 }
 
-function* createFilter(action: CreateFilterAction) {
+function* createFilter(
+  action: ActionWithPayload<
+    CreateFilterActionTypes.CREATE_FILTER_REQUEST,
+    CreateFilterPayload
+  >
+) {
```

Comment:
@EDsCODE I seem to have come up with a solution - lmk what you think! Would be good if it could be slightly cleaner.

Reply:
there may be a way that you can extract the request type specifically but I can't think of it atm. otherwise, this looks right!